src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.cs

```diff
@@ -102,6 +102,19 @@ private protected override JsonPropertyInfo CreatePropertyInfoForTypeInfo()
             };
         }
 
+        private protected sealed override JsonPropertyInfo CreateJsonPropertyInfo(JsonTypeInfo declaringTypeInfo, JsonSerializerOptions options)
+        {
+            // Options on this type info might not be the same as the one passed in
```

Comment:
For consistency, shouldn't we just be passing `declaringTypeInfo.Options` to the child property?